_posts/2017-10-17-faf-in-openshift.markdown

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title:  "FAF in OpenShift"
+date:   2017-10-17 12:00:00 +0100
+author: mkutlak
+visible: 1
+categories: FAF container OpenShift
+---
+
+Guide on how to deploy your own [FAF](https://www.github.com/abrt/faf) in [OpenShift](https://www.openshift.com/).
+
+# Deploy FAF in OpenShift
+
+The following text will guide you through the process of deploying FAF in OpenShift and using it to collect crash reports from your machines.
+There are two options how you can do it. The first option is to use the command line, and the second one is to use the web interface.
+
+## Prerequisites
+ - [OpenShift Origin](https://github.com/openshift/origin/releases) or [OpenShift Container Platform](https://www.openshift.com/).
+ - Cluster admin rights for editing security context constraints, and managing
+   volumes.
+
+
+## Command Line
+
+1. **Create a new project.**
+```
+oc new-project <project_name>
+```
+
+2. **Add the FAF application.** <br>
+We've created docker image with FAF, that you can find on
+[Docker Hub](https://hub.docker.com/r/abrt/faf-image).
+```
+oc new-app abrt/faf-image --name <app_name>
+```
+This command will pull the image from docker hub and then deploy it.
+
+3. **Edit Security Context Constraints.** <br>
+The deployment won't succeed because the FAF container requires usage of the root user, but default
```

Comment:
Hmm. Why it needs root?

Reply:
It needs root for [run_faf](https://github.com/abrt/faf/blob/master/docker/files/run_faf) script.